cloudinit/sources/DataSourceVMware.py

```diff
@@ -131,14 +173,33 @@ class DataSourceVMware(sources.DataSource):
     def __init__(self, sys_cfg, distro, paths, ud_proc=None):
         sources.DataSource.__init__(self, sys_cfg, distro, paths, ud_proc)
 
+        self.cfg = {}
         self.data_access_method = None
         self.vmware_rpctool = VMWARE_RPCTOOL
 
+        # A list includes all possible data transports, each tuple represents
+        # one data transport type. This datasource will try to get data from
+        # each of transports follows the tuples order in this list.
+        # A tuple has 3 elements which are:
+        # 1. The transport name
+        # 2. The function name to get data for the transport
+        # 3. A boolean tells whether the transport requires VMware platform
+        self.possible_data_access_method_list = [
```

Comment:
See earlier comment regarding order.

Reply:
updated the order, put imc to the tail of list.